app/_how-tos/store-and-use-your-mistral-api-key-as-a-secret-in-konnect-config-store.md

```diff
@@ -0,0 +1,191 @@
+
+
+
+<!--vale off-->
+{% control_plane_request %}
+url: /v2/control-planes/$CONTROL_PLANE_ID/core-entities/vaults/
+status_code: 201
+method: POST
+headers:
+    - 'Accept: application/json'
+    - 'Content-Type: application/json'
+    - 'Authorization: Bearer $KONNECT_TOKEN'
+body:
+    config:
+      config_store_id: $CONFIG_STORE_ID
+    description: Storing secrets in Konnect
+    name: konnect
+    prefix: mysecretvault
+{% endcontrol_plane_request %}
+<!--vale on-->
+
+## 3. Store the Mistral AI key as a secret
+
+In this tutorial, you'll be storing the Mistral API key you set previously and using it to generate an answer to a question using the [AI Proxy plugin](/plugins/ai-proxy/). By storing it as a secret in a {{site.konnect_short_name}} Vault, you can reference it in the plugin configuration in the next step.
+
+Store your Mistral key as a secret by sending a `POST` request to the `/v2/control-planes/{controlPlaneId}/config-stores/{configStoreId}/secrets` endpoint:
+
+<!--vale off-->
+{% control_plane_request %}
+url: /v2/control-planes/$CONTROL_PLANE_ID/config-stores/$CONFIG_STORE_ID/secrets/
+status_code: 201
+method: POST
+headers:
+    - 'Accept: application/json'
+    - 'Content-Type: application/json'
+    - 'Authorization: Bearer $KONNECT_TOKEN'
+body:
+    key: mistral-key
+    value: Bearer <mistral-key-here>
+{% endcontrol_plane_request %}
+<!--vale on-->
+
+## 4. Reference your stored Mistral API key secret
+
+To reference your stored Mistral API key secret, you use the prefix from your Vault config, the name of the secret, and optionally the property in the secret you want to use. Now, you'll reference the Mistral API key as a secret in the authorization header of the AI Proxy plugin configuration.
+
+Enable the AI Proxy plugin on your route by sending a `POST` request to the [`/routes/{RouteId}/plugins` endpoint](/api/konnect/control-planes-config/v2/#/operations/create-plugin-with-route):
+
+<!--vale off-->
+{% control_plane_request %}
+url: /v2/control-planes/$CONTROL_PLANE_ID/core-entities/plugins
+status_code: 201
+method: POST
+headers:
+    - 'Accept: application/json'
+    - 'Content-Type: application/json'
+    - 'Authorization: Bearer $KONNECT_TOKEN'
+body:
+    config:
+      route_type: llm/v1/chat
+      auth:
+        header_name: Authorization
+        header_value: '{vault://mysecretvault/mistral-key}'
+      model:
+        provider: mistral
+        name: mistral-tiny
+        options: 
+          mistral_format: openai
+          upstream_url: https://api.mistral.ai/v1/chat/completions
+    enabled: true
+    name: ai-proxy
+    route:
+      id: $ROUTE_ID
+{% endcontrol_plane_request %}
+<!--vale on-->
+
+## 5. Validate
+
+To verify that {{site.base_gateway}} can pull the secrets from the {{site.konnect_short_name}} Config Store, you can use the AI Proxy plugin to confirm that the plugin is using the correct API key when a request is made:
```

Comment:
```suggestion
You can use the AI Proxy plugin to confirm that the plugin is using the correct API key when a request is made:
```
We are testing if Mistral is properly working not and that proves that the secrets were referenced.